File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // initial prompt
    printf("O hai! How much change is owed?\n");
    double change = get_float();
    
    // continuously ask user for change if input is negative
    while (change < 0.0) {
        printf("How much change is owed?\n");
        change = get_float();
    }
    
    change = round(change * 100);
    int coins = 0;
    
    while (change >= 25.0)
    {
        change -= 25.0;
        coins++;
    }
    
    while (change >= 10.0)
    {
        change -= 10.0;
        coins++;
    }

    while (change >= 5.0)
    {
        change -= 5.0;
        coins++;
    }

    while (change >= 1.0)
    {
        change -= 1;
        coins++;
    }
    
    // print the number of coins
    printf("%i\n", coins);
}

File: pset6/greedy.py
import cs50

while True:
    print("How much change is owed?")
    change = cs50.get_float()
    if change > 0:
        break

# convert the input to an integer
change = round(change * 100)

# calculate quantity of each type of coin
q = change // 25
d = change % 25 // 10
n = change % 25 % 10 // 5
p = change % 25 % 10 % 5 // 1

# print the sum of all the coin quantities
print(q + d + n + p)

exit(0)